pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator_helper.go

```diff
@@ -32,13 +33,19 @@ import (
 func (c *K8sOrchestrator) getPVCAnnotations(ctx context.Context, volumeID string) (map[string]string, error) {
 	log := logger.GetLogger(ctx)
 	log.Debugf("Getting annotations on pvc corresponding to volume: %s", volumeID)
+	pvcAnn := make(map[string]string)
 	if pvc := c.volumeIDToPvcMap.get(volumeID); pvc != "" {
 		parts := strings.Split(pvc, "/")
 		pvcNamespace := parts[0]
 		pvcName := parts[1]
 
 		pvcObj, err := c.informerManager.GetPVCLister().PersistentVolumeClaims(pvcNamespace).Get(pvcName)
 		if err != nil {
+			if apierrors.IsNotFound(err) {
+				//PVC may have been deleted. Return.
+				log.Debugf("PVC %s is not found in namespace %s using informer manager", pvcName, pvcNamespace)
+				return pvcAnn, nil
```

Comment:
Instead of sending empty pvcAnn, how about we throw `PVCNotFound` error from here? The ClearFakeAttached can check if the error is PVCNotFound, if yes, it can ignore clearing the annotation.

Reply:
+1

Reply:
Makes sense. Changed.